packages/runtime/container-runtime/src/containerRuntime.ts

```diff
@@ -1616,8 +1617,19 @@ export class ContainerRuntime
 			return this.submitSignalFn(envelope2, targetClientId);
 		};
 
+		let snapshot: ISnapshot | ISnapshotTree | undefined = getSummaryForDatastores(
```

Comment:
This being two different types and undefined makes this a little weird. Is there a way we can simplify this?

Reply:
Actually, it was ISnapshotTree | undefined before and this is kind of first step towards probably moving to ISnapshot where ISnapshpt.snapshotTree would be this current snapshotTree. We can pass this down to layers and they can themselves make decisions whether to fetch the snapshot or not due to missing blob contents and they don't have to reach out to container runtime to ask that question. Also, we are passing down the blobs, so even for blob reads they don't need to access storage as then they would have everything, and they can then just load.